public/app/plugins/datasource/cloudwatch/datasource.ts

```diff
@@ -906,9 +898,7 @@ export class CloudWatchDatasource
     return queries.map((query) => ({
       ...query,
       region: this.getActualRegion(this.replace(query.region, scopedVars)),
-      expression: this.replace(query.expression, scopedVars),
-
-      ...(!isCloudWatchLogsQuery(query) && this.interpolateMetricsQueryVariables(query, scopedVars)),
+      ...(isCloudWatchMetricsQuery(query) && this.interpolateMetricsQueryVariables(query, scopedVars)),
```

Comment:
is this valid syntax? are you expanding `false` if it's not a metrics query?

Reply:
yeah the spread operators requires an interable. since `boolean` is not it does work. although it's not very pretty, refactoring this is probably out of scope of this PR.